add-ons/shadow/pkg/handler/fake/fake.go

```diff
@@ -0,0 +1,282 @@
+  annotations:
+    mesh.megaease.com/service-labels: version=shadow
+    mesh.megaease.com/service-name: visits-service
+    mesh.megaease.com/shadow-service-name: shadow-visits-service
+  labels:
+    mesh.megaease.com/shadow-service: "true"
+  name: visits-service-shadow
+  namespace: spring-petclinic
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: visits-service
+      mesh.megaease.com/shadow-service: "true"
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        app: visits-service
+        mesh.megaease.com/shadow-service: "true"
+    spec:
+      initContainers: []
+      containers:
+      - args:
+        - -c
+        - java -server -Xmx1024m -Xms1024m -Dspring.profiles.active=sit -Djava.security.egd=file:/dev/./urandom  org.springframework.boot.loader.JarLauncher
+        command:
+        - /bin/sh
+        env:
+        - name: JAVA_TOOL_OPTIONS
+          value: ' -javaagent:/agent-volume/easeagent.jar -Deaseagent.log.conf=/agent-volume/log4j2.xml '
+        image: megaease/spring-petclinic-visits-service:latest
+        imagePullPolicy: Always
+        lifecycle:
+          preStop:
+            exec:
+              command:
+              - sh
+              - -c
+              - sleep 10
+        name: visits-service
+        ports:
+        - containerPort: 8080
+          protocol: TCP
+        resources:
+          limits:
+            cpu: "2"
+            memory: 1Gi
+          requests:
+            cpu: 200m
+            memory: 256Mi
+        volumeMounts:
+        - mountPath: /application/application-sit.yml
+          name: configmap-volume-0
+          subPath: application-sit.yml
+      volumes:
+      - configMap:
+          items:
+          - key: application-sit-yml
+            path: application-sit.yml
+          name: visits-service
+        name: configmap-volume-0
+`
+
+// NewServiceCanary create fake ServiceCanary for test.
+func NewServiceCanary() *resource.ServiceCanary {
+	decoder := k8Yaml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(fakeCanaryYaml)), 1000)
+	serviceCanary := &resource.ServiceCanary{}
+	_ = decoder.Decode(serviceCanary)
+	return serviceCanary
+}
+
+// NewDeletedServiceCanary create fake ServiceCanary for test.
+func NewDeletedServiceCanary() *resource.ServiceCanary {
+	decoder := k8Yaml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(fakeDeleteCanaryYaml)), 1000)
+	serviceCanary := &resource.ServiceCanary{}
+	_ = decoder.Decode(serviceCanary)
+	return serviceCanary
+}
+
+// NewShadowService create fake ShadowService for test.
+func NewShadowService() object.ShadowService {
+	shadowService := object.ShadowService{
+		Name:        "shadow-visits-service",
+		ServiceName: "visits-service",
+		Namespace:   "spring-petclinic",
+	}
+	return shadowService
+}
+
+// NewNameSpace create fake NameSpace for test.
+func NewNameSpace() *corev1.Namespace {
```

Comment:
`NameSpace` should be `Namespace` to align with `corev1.Namespace`.

```suggestion
// NewNamespace create fake Namespace for test.
func NewNamespace() *corev1.Namespace {
```